packages/kbn-eslint-plugin-packages/src/rules/no_naked_eslint_disable.test.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import dedent from 'dedent';
+import { RuleTester } from 'eslint';
+import { NoNakedESLintDisableRule } from './no_naked_eslint_disable';
+
+const fmt = (str: TemplateStringsArray) => dedent(str);
+
+const tsTester = [
+  '@typescript-eslint/parser',
+  new RuleTester({
+    parser: require.resolve('@typescript-eslint/parser'),
+    parserOptions: {
+      sourceType: 'module',
+      ecmaVersion: 2018,
+      ecmaFeatures: {
+        jsx: true,
+      },
+    },
+  }),
+] as const;
+
+const babelTester = [
+  '@babel/eslint-parser',
+  new RuleTester({
+    parser: require.resolve('@babel/eslint-parser'),
+    parserOptions: {
+      sourceType: 'module',
+      ecmaVersion: 2018,
+      requireConfigFile: false,
+      babelOptions: {
+        presets: ['@kbn/babel-preset/node_preset'],
+      },
+    },
+  }),
+] as const;
+
+for (const [name, tester] of [tsTester, babelTester]) {
+  describe(name, () => {
+    tester.run('@kbn/packages/no_naked_eslint_disable', NoNakedESLintDisableRule, {
+      valid: [
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            // eslint-disable no-var
+          `,
+        },
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            // eslint-disable-next-line no-use-before-define
+          `,
+        },
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            // eslint-disable-line no-use-before-define
+          `,
+        },
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            /* eslint-disable no-var */
+          `,
+        },
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            /* eslint-disable no-console, no-control-regex*/
+          `,
+        },
+      ],
+
+      invalid: [
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            /* eslint-disable */
+            const a = 1;
+          `,
+          errors: [
+            {
+              line: 1,
+              message:
+                'Using a naked eslint disable is not allowed. Please specify the specific rules to disable.',
+            },
+          ],
+          output: '\nconst a = 1;',
+        },
+        {
+          filename: 'foo.ts',
+          code: fmt`
+            // eslint-disable-next-line
```

Comment:
I think we should expand the tests to handle comments after and in the middle of code too.

Reply:
👍